### DISCUSSION TITLE
Server Jamulus on Mac

### DISCUSSION THREAD

**Discussion by ayamel64** [jamulussoftware/jamulus#3420] [Bug Investigation]
Hi, 
I wanted to point out that the "none" function of the Jamulus server doesn't work on MAC. In practice, unlike Windows where this option works very well, on Mac by setting "none" once you enter the server, anyone who searches sees it anyway. Thank you. HI

**Comment by mcfnord** [comment]
Interesting. Which directory does the server appear on? The one it was on last?

**Comment by ayamel64** [comment]
I honestly didn't notice
Sorry
Daniele
Inviato da iPad

> Il giorno 9 nov 2024, alle ore 20:22, John Dempsey ***@***.***> ha scritto:
> 
> Interesting. Which directory does the server appear on? The one it was on last?
>

**Comment by ann0see** [comment]
Smells like a bug to me.

**Comment by ayamel64** [comment]
It doesn't stink at all. !!The jamulus server on Mac launching it in standard mode once all the musicians entered the room and then setting it to “none” or custom so as to become private , IT DOES NOT WORK .From Windows instead it works.I'm not here to waste time or make others lose.Try it and you will see . Thank youInviato da iPadIl giorno 10 nov 2024, alle ore 23:39, ann0see ***@***.***> ha scritto:﻿
Smells like a bug to me.

**Comment by chrisrimple** [comment]
@ayamel64 "Smells like a bug" is a common phrase in America that means "I think this is a bug".  In other words, @ann0see is agreeing with you that there may be a problem.

I have only run headless servers, not using the GUI.  So this is just a guess, but doesn't the value NONE need to be set before launching the server?  Said another way, if you launch the Server app on Mac, I assume that it immediately reports itself to the directory server, so it's visible.  If you then change it to NONE, you need to quit and restart the Server app, so the setting is saved and then used on re-launch.  I suspect the same is true on Windows.  But again, just a guess - if I'm right, there's no bug.  This could be tested by launching the Server app, setting NONE, quitting, waiting 15 minutes, and launching the Server app again to see if it's visible in the directory.  If it is visible, then the feature is not working correctly.

**Reply by ann0see** [reply]
> "Smells like a bug" is a common phrase in America that means "I think this is a bug". 

Yes. Exactly. 



It should definitely be possible to first register and then de register from the macOS GUI. 

Which Jamulus version do you run?

**Comment by ayamel64** [comment]
Thanks for the clarification 😉Unfortunately in Windows we did theTests and it works.If you start a server for example on genres 1 and I call it Goofy andMy friends go inside, then go to theSettings and setting "none" anyone who goes on generates one the Pippo Server does not see it.Look, even on the MAC I've always done this and it worked.Now, however, it doesn't go anymore and everyone comes in being visible.Thank youDanielInviato da iPhoneIl giorno 11 nov 2024, alle ore 18:24, Chris Rimple ***@***.***> ha ***@***.*** "Smells like a bug" is a common phrase in America that means "I think this is a bug".  In other words, @ann0see is agreeing with you that there may be a problem.
I have only run headless servers, not using the GUI.  So this is just a guess, but doesn't the value NONE need to be set before launching the server?  Said another way, if you launch the Server app on Mac, I assume that it immediately reports itself to the directory server, so it's visible.  If you then change it to NONE, you need to quit and restart the Server app, so the setting is saved and then used on re-launch.  I suspect the same is true on Windows.  But again, just a guess - if I'm right, there's no bug.  This could be tested by launching the Server app, setting NONE, quitting, waiting 15 minutes, and launching the Server app again to see if it's visible in the directory.  If it is visible, then the feature is not working correctly.

**Reply by rdica** [reply]
I just tested 3.11.0 server on OSX Sonoma 14.7.1.
I registered on each directory, including a private one I have.
Disconnected. Reconnected/disconnected. Each time I see it appear in explorer.jamulus.io, then disappear when I select None for Genre. Behaves as expected.
